units/periodic_builds.go

```diff
@@ -70,39 +75,55 @@ func (j *periodicBuildJob) Run(ctx context.Context) {
 		j.AddError(errors.New("project not found"))
 		return
 	}
-
-	catcher := grip.NewBasicCatcher()
-	for _, definition := range j.project.PeriodicBuilds {
-		shouldRerun, err := j.shouldRerun(definition)
-		if err != nil {
-			catcher.Add(err)
-			continue
-		}
-		if shouldRerun {
-			catcher.Add(j.addVersion(ctx, definition))
+	var definition *model.PeriodicBuildDefinition
+	for _, d := range j.project.PeriodicBuilds {
+		if d.ID == j.DefinitionID {
+			definition = &d
+			break
 		}
 	}
+	if definition == nil {
+		j.AddError(errors.New("no definition ID found"))
+		return
+	}
+	defer func() {
```

Comment:
Should this be higher? If there's a transient db error that causes lines 72 or 76 to return, then will the periodic build get rescheduled, or would we have to manually intervene?

Reply:
If one of those errors are hit, the job generator will actually cause us to retry this immediately (since it creates this job if the timestamp is in the past)